chainer/initializers/constant.py

```diff
@@ -30,17 +30,22 @@ def __call__(self, array):
         if len(shape) != 2 or shape[0] != shape[1]:
             raise ValueError('Identity matrix initialization can only be used '
                              'for 2D squared matrices.')
-        array[...] = 0
-        xp = backend.get_array_module(array)
-        xp.fill_diagonal(array, self.scale)
+
+        device = backend.get_device_from_array(array)
+        # ideep64 does not support fill_diagonal
+        if isinstance(device, chainer.backends.intel64.Intel64Device):
```

Comment:
How about `array[...] = device.xp.identity(shape[0]) * self.scale` which should work in any case?

Reply:
Doesn't it introduce an extra allocation & memory copy?

Reply:
Doesn't it introduce an extra allocation & memory copy?

Reply:
Yes, but I don't think that's a problem because initializers are not called so often.

Reply:
Ok!